### ISSUE TITLE
[Feature request] Support for blocks in pre_tasks.

### ISSUE THREAD

**Issue by coxoperationsengineer** [haidaraM/ansible-playbook-grapher#95]
**Is your feature request related to a problem? Please describe.**
I use `blocks` in almost every part of a `play` where a `block` is allowed, including `pre_tasks`, but it seems that the grapher is unable to handle those, and generates an `exception`, making it unusable in most of my complex playbooks :(  I wasn't sure if this was a bug, or a feature request, so pardon me if it is the former, and not the latter.

```bash
Traceback (most recent call last):
  File "/Users/myUserName/.pyenv/versions/venvGraphAnsible/bin/ansible-playbook-grapher", line 8, in <module>
    sys.exit(main())
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/cli.py", line 126, in main
    cli.run()
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/cli.py", line 43, in run
    playbook_node = parser.parse()
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/parser.py", line 154, in parse
    self._include_tasks_in_blocks(current_play=play, parent_nodes=[play_node], block=pre_task_block,
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/parser.py", line 245, in _include_tasks_in_blocks
    parent_nodes[-1].add_node(f"{node_type}s", EdgeNode(parent_nodes[-1], role_node,
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/graph.py", line 58, in add_node
    raise Exception(
Exception: The target composition 'pre_tasks' is unknown. Supported are: ['tasks']
```

**Describe the solution you'd like**
The same support as found when using `blocks` in `tasks`

**Describe alternatives you've considered**
N/A

**Additional context**
Love the library when I can use it, but on my complex `plays` where I use `blocks` to keep them clean and apply conditionals on large groups of `tasks`, where I would love to see the graph to explain to people what is going on, it is a no go.

**Comment by haidaraM** [comment]
Thanks for reporting the issue. I reproduced it locally. I will make a release soon

**Comment by haidaraM** [comment]
Can you test with the latest dev release: `pip install ansible-playbook-grapher==1.0.1.dev0`

**Comment by coxoperationsengineer** [comment]
@haidaraM sure!

I get the following on 1.0.1.dev0, I tried two different playbooks that are both setup very similar in regards to the `pre_tasks`, both result in same `exception`.   Let me know what else you would like me to test, would love to help straighten it out.

```bash
ansible-playbook-grapher nist_dsr_link_launch.yml -i inventory/hosts --vault-password-file $VAULT --include-role-tasks -e '@dsr_play_args.yml' --view
[WARNING]: ansible.utils.display.initialize_locale has not been called, this may result in incorrectly calculated text widths that can cause Display to print incorrect line lengths

Parsing Play: all (805) ********************************************************************************************************************************************************************************


Done parsing Play: all (805) ***************************************************************************************************************************************************************************


Parsing Play: all (805) ********************************************************************************************************************************************************************************
Traceback (most recent call last):
  File "/Users/myUserName/.pyenv/versions/venvGraphAnsible/bin/ansible-playbook-grapher", line 8, in <module>
    sys.exit(main())
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/cli.py", line 126, in main
    cli.run()
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/cli.py", line 43, in run
    playbook_node = parser.parse()
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/parser.py", line 154, in parse
    self._include_tasks_in_blocks(current_play=play, parent_nodes=[play_node], block=pre_task_block,
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/parser.py", line 284, in _include_tasks_in_blocks
    self._include_tasks_in_blocks(current_play=current_play, parent_nodes=parent_nodes, block=b,
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/parser.py", line 230, in _include_tasks_in_blocks
    self._include_tasks_in_blocks(current_play=current_play, parent_nodes=parent_nodes, block=task_or_block,
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/parser.py", line 308, in _include_tasks_in_blocks
    self._add_task(task=task_or_block, task_vars=play_vars, node_type=node_type,
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/parser.py", line 86, in _add_task
    parent_node.add_node(target_composition=f"{node_type}s", node=edge_node)
  File "/Users/myUserName/.pyenv/versions/3.10.0/envs/venvGraphAnsible/lib/python3.10/site-packages/ansibleplaybookgrapher/graph.py", line 58, in add_node
    raise Exception(
Exception: The target composition 'pre_tasks' is unknown. Supported are: ['tasks']
```

**Comment by haidaraM** [comment]
Can you send me a minimal example that is failing and the output of `ansible-playbook-grapher --version`?

**Comment by coxoperationsengineer** [comment]
Yep, I’ll make a copy of my playbook and try to rip out some pieces and see
if I can narrow it down as well.  The challenge is that nothing about these
plays are minimal, is there a more verbose mode where I can see what
section it is getting to and failing on?

Also noticed that ‘ansible.utils’ error about locale, not sure what that is
about but I’m using the latest version of that collection.

**Comment by haidaraM** [comment]
Yes, there is a more verbose mode: `ansible-playbook-grapher -vvvv ...`. Can you send me the output with this flag ?

Are you using some includes or imports inside your pre_tasks ?

**Comment by coxoperationsengineer** [comment]
The `verbose` mode helped track it down further!

When I comment out the `pre_tasks` the tool does run with no exceptions, but gives me this error, and the resulting graph line headings and identifiers are all misaligned.
```bash
[WARNING]: ansible.utils.display.initialize_locale has not been called, this may result in incorrectly
calculated text widths that can cause Display to print incorrect line lengths
```

I am using includes and imports in `pre_tasks`, and that appears to be where it fails, it hits my `pre_tasks`, which does an include of `analyze`, inside `analyze`, the `main.yml` does `import` of 3 task files from the `analyze` role, and it fails when it hits that first task file, which also has a `block`.

Here is the exception and setup when `pre_tasks` is in the `play`:

```bash
Parsing Play: all (805) *********************************************************************************
Parsing pre_tasks...
Adding pre_task 'Verify Vars Needed For This Playbook At Initial Launch Are Present.' to the graph
Adding pre_task 'Verify Reachability and Authentication, Or FAIL The Host.' to the graph
Adding pre_task 'pre_task_work : Assert The User Provided Email Address Was Formatted Correctly As A List Or Fail The Host.' to the graph
Adding pre_task 'pre_task_work : Assert The User Provided Email Addresses Do Not Contain White Space And Are Addressed To @.com Domains Only Or Fail The Host.' to the graph
Adding pre_task 'pre_task_work : Set Fact Containing The User Provided Email List Plus default address WHEN: user_email_list > 0, ELSE Skipped.' to the graph
Adding pre_task 'pre_task_work : Set Fact Containing The default address Email WHEN: user_email_list == 0, ELSE Skipped.' to the graph
An 'include_role' found. Including tasks from the role 'analyze'
statically imported: myProjectDir/roles/analyze/tasks/analyze_cpm_filter_status.yml
Loading collection ansible.utils from myProjectDir/collections/ansible_collections/ansible/utils
statically imported: myProjectDir/roles/analyze/tasks/analyze_ip_filter_action.yml
statically imported: myProjectDir/roles/analyze/tasks/analyze_ipv6_filter_action.yml
Adding pre_task 'analyze : Gather Current Configured CPM State.' to the graph
Traceback (most recent call last):
  File "/Users/myUserName/.pyenv/versions/venvGraphAnsible/bin/ansible-playbook-grapher", line 8, in <module>
    sys.exit(main())
```

`role/tasks/analyze/main.yml`
```yaml
- name: 'Include Validate CPM filter State.'
  ansible.builtin.import_tasks: analyze_cpm_filter_status.yml

- name: 'Include: Validate IP Filters And Associated Actions.'
  ansible.builtin.import_tasks: analyze_ip_filter_action.yml
  when:
    - inventory_hostname not in groups['mtso']

- name: 'Include: Validate IPv6 Filters And Associated Actions.'
  ansible.builtin.import_tasks: analyze_ipv6_filter_action.yml
  when:
    - inventory_hostname in groups['mgmtspe']
```

`analyze_cpm_filter_status.yml`
```yaml
- name: 'Analyzing CPM Filter State(s).'
  block:
    - name: 'Gather Current Configured CPM State.'
      ansible.netcommon.cli_command:
        command: show system security cpm-filter ip-filter
      connection: ansible.netcommon.network_cli
      register: nokia_show_system_security_cpmfilter_ipfilter
```

**Comment by haidaraM** [comment]
Can you try the latest version on the branch fix/blocks with ` pip install git+https://github.com/haidaraM/ansible-playbook-grapher@fix/blocks`?

The warning on display should have no impact on the generated graph. Can you share a screenshot with the misaligned headlings ?

**Comment by coxoperationsengineer** [comment]
Same result with the new pip installed library version from Git, do you have an email address I can send the screenshots to?  I would prefer not to have some of the content out in the open.

**Comment by haidaraM** [comment]
Hmm there should be another case I'm not testing at the moment... Can you just uninstall and start from scratch with the latest version on the branch ? Just to be sure

My email: elmhaidara@gmail.com

**Comment by coxoperationsengineer** [comment]
Emailed you @haidaraM

**Comment by haidaraM** [comment]
Just saw your mail and replied to it. So a complete reinstall fixed the main issue. For the rendering issue with big playbooks and large number of tasks, I will create another issue.

**Comment by haidaraM** [comment]
Released in 1.0.1
